internal/provisioner/kops_provisioner.go

```diff
@@ -298,14 +300,21 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, aws aw
 	}
 
 	// Get the new ELB internal-nginx endpoint
-	endpoint, err := getLoadBalancerEndpoint("internal-nginx", logger, kops.GetKubeConfigPath())
+	logger.Infof("Waiting up to %d seconds for internal ELB to be ready...", wait)
+	defer cancel()
+	endpoint, err := getLoadBalancerEndpoint(ctx, "internal-nginx", logger, kops.GetKubeConfigPath())
 	if err != nil {
 		return err
 	}
 
+	if endpoint == "" {
+		logger.Errorf("ELB endpoint is empty...")
```

Comment:
We don't really need to log the error and return it. It will be logged when it is returned.